tools/clang/lib/CodeGen/CGDecl.cpp

```diff
@@ -888,6 +894,11 @@ llvm::Value *CodeGenFunction::EmitLifetimeStart(uint64_t Size,
 }
 
 void CodeGenFunction::EmitLifetimeEnd(llvm::Value *Size, llvm::Value *Addr) {
+  // HLSL Change Begins
+  // Don't emit the intrinsic for hlsl for now unless it is explicitly enabled
+  if (!CGM.getCodeGenOpts().HLSLEnableLifetimeMarkers)
+    return;
+  // HLSL Change Ends
```

Comment:
I'll comment here just to let you know I left this out of the patch since it's not required - if no lifetime.start intrinsics are generated, this is never executed (the original code had only start intrinsics disabled, too). Same for the change in CGHLSLMS.cpp below.

Reply:
I trust your judgement, but in my testing it definitely appeared without a start at one point. Perhaps later changes I made eliminated that possibility.

Reply:
The most likely reason for that would be the inlining case that this commit doesn't cover, so I'm pretty confident this is the right thing to do. In all my testing during development of the lifetimes branch I've always gone back to "top of tree behavior" for comparison by modifying two places: returning nullptr in CGDecl when emitting start markers and passing 'false' to the AlwaysInliner.
I'm making sure I have all tests covered, actually ran into a really weird one with DeadLoopFail that I'm currently bisecting...

Reply:
Actually, I just looked at this again and stand partially corrected: This one here is not necessary, but the one in CGHLSLMS.cpp is (there's actually an EmitLifetimeStart a few lines up that also needs to be guarded).